src/classes/OPP_StageMappingUtil.cls

```diff
@@ -91,10 +95,198 @@ public with sharing class OPP_StageMappingUtil {
+
+                    String oppState = mappingRecord.Opportunity_State__c.trim().toUpperCase();
+                    String oppStage = mappingRecord.Opportunity_Stage__c.trim().toLowerCase();
+                    Set<String> stages = oppStateStages.containsKey(oppState) ? oppStateStages.get(oppState) : new Set<String>();
+                    stages.add(oppStage);
+                    oppStateStages.put(oppState, stages);
+                }
+            }
+            
+            COMMITTED_STATE_SET = oppStateStages.containsKey(COMMITTED_STATE_VAL) ? oppStateStages.get(COMMITTED_STATE_VAL) : new Set<String>();
+            FINALIZED_STATE_SET = oppStateStages.containsKey(FINALIZED_STATE_VAL) ? oppStateStages.get(FINALIZED_STATE_VAL) : new Set<String>();
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Returns a list of Opportunity stage to state mappings. If it is a test running,
+    * return an artificially created list. Otherwise, get it from the database
+    ********************************************************************************************************/
+    private static List<Opportunity_Stage_To_State_Mapping__mdt> getOpportunityStageToStateMappings() {
+        List<Opportunity_Stage_To_State_Mapping__mdt> metadataRecordList = new List<Opportunity_Stage_To_State_Mapping__mdt>();
+
+        if (Test.isRunningTest()) {
+            
+            // If it's a test running, return artificial records. This ensures consistency in the testing. 
+            Opportunity_Stage_To_State_Mapping__mdt finalizedRecord = new Opportunity_Stage_To_State_Mapping__mdt();
+            finalizedRecord.Opportunity_Stage__c = 'Closed Won';
+            finalizedRecord.Opportunity_State__c = FINALIZED_STATE_VAL;
+
+            Opportunity_Stage_To_State_Mapping__mdt pledgedRecord = new Opportunity_Stage_To_State_Mapping__mdt();
+            pledgedRecord.Opportunity_Stage__c = 'Pledged';
+            pledgedRecord.Opportunity_State__c = COMMITTED_STATE_VAL;
+
+            metadataRecordList.add(finalizedRecord);
+            metadataRecordList.add(pledgedRecord);
+        } else {
+            metadataRecordList = [Select Id, Opportunity_Stage__c, Opportunity_State__c FROM Opportunity_Stage_To_State_Mapping__mdt];
+        }
+
+        return metadataRecordList;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates the finalized state record if there are no finalized records. It utilizes the 
+    * "Opportunity Stage when Fully Paid" field as the Opportunity stage for the finalized state. 
+    ********************************************************************************************************/
+    public static List<Metadata.CustomMetadata> createFinalizedOppStageToStateRecord(List<Metadata.CustomMetadata> metadataToInsert) {
```

Comment:
@jbrandolinisf The description for this class is `API Class to allow caller to determine Opportunity's State (Uncommitted, Committed. or Finalized) using the Opportunities Stage`. Given the single responsibility for this class, would it make sense to move all of the methods related to creating and deploying CMT's for this custom object into a different class such as `OPP_StageMappingCMT` or `OPP_StageMappingConfigBuilder`?

Reply:
@force2b - Yep, Kiet and I talked about it.  Originally I was thinking we would add other support to this class (to add/remove stage to state mappings, but I am ok with moving this to another class, and @KietTruongSFDO is doing this now.

Reply:
I moved it into a new class named "OPP_StageMappingCMT"